checks/tools/io/nco.py

```diff
@@ -2,206 +2,217 @@
-        # NOTE: we only verify that there is no error produced. We do not
-        # verify the resulting file. Verifying would not be straight forward
-        # as the following does not work (it seems that there is a timestamp
-        # of file creation inserted in the metadata or similar):
-        # diff test_echam_spectral - deflated_wl.nc4 \
-        #      test_echam_spectral - deflated_wind10_wl.nc4
-        self.sanity_patterns = sn.all([
-            sn.assert_not_found(r'(?i)unsupported|error', self.stderr),
-            sn.assert_not_found(r'(?i)unsupported|error', self.stdout)
-        ])
-
-
-@rfm.simple_test
-class NCO_MergeNC4CTest(NCOBaseTest):
-    def __init__(self):
-        super().__init__()
-        self.descr = ('verifies merging and compressing of two compressed '
-                      'netCDF-4 files')
-        self.executable = 'ncks'
-        self.executable_opts = ['-A', '-L', '2',
-                                'test_echam_spectral-deflated_wind10.nc4c',
-                                'test_echam_spectral-deflated_wl.nc4c']
-        # NOTE: we only verify that there is no error produced. We do not
-        # verify the resulting file. Verifying would not be straight forward
-        # as the following does not work (it seems that there is a timestamp
-        # of file creation inserted in the metadata or similar):
-        # diff test_echam_spectral-deflated_wl.nc4c \
-        #      test_echam_spectral-deflated_wind10_wl.nc4c
-        self.sanity_patterns = sn.all([
-            sn.assert_not_found(r'(?i)unsupported|error', self.stderr),
+    test_type = parameter(['nc', 'nc4', 'nc4c'])
+    descr = ('verifies merging of 2 netCDF files')
+    # NOTE: we only verify that there is no error produced. We do not
+    # verify the resulting file. Verifying would not be straight forward
+    # as the following does not work (it seems that there is a timestamp
+    # of file creation inserted in the metadata or similar):
+    # diff sresa1b_ncar_ccsm3-example_tas.nc \
+    #      sresa1b_ncar_ccsm3-example_pr_tas.nc
+
+    @run_before('run')
+    def copy_input_data(self):
+        inputs = {
+            'nc': ['sresa1b_ncar_ccsm3-example_pr.nc',
+                   'sresa1b_ncar_ccsm3-example_tas.nc'],
+            'nc4': ['test_echam_spectral-deflated_wind10.nc4',
+                    'test_echam_spectral-deflated_wl.nc4'],
+            'nc4c': ['test_echam_spectral-deflated_wind10.nc4c',
+                     'test_echam_spectral-deflated_wl.nc4c'],
+        }
+        input_path = os.path.join(self.current_system.resourcesdir, 'CDO-NCO')
+        for idx in range(2):
+            self.prerun_cmds += [
+                f'cp {input_path}/{inputs[self.test_type][idx]} .']
+
+    @run_before('run')
+    def setup_per_partition_name(self):
+        self.executable = (
+            f'bash {self.squashfs_script} ncks'
+            if 'squashfs' in self.current_partition.name else 'nco')
```

Comment:
```suggestion
            if 'squashfs' in self.current_partition.name else 'ncks')
```
This should be ncks, right?

Reply:
https://github.com/eth-cscs/cscs-reframe-tests/pull/34/commits/9b382cddca9e2d926eb1316103ce249618f24bf0